source/Octopus.Tentacle.Tests.Integration/Support/TentacleConfigurationsAttribute.cs

```diff
@@ -0,0 +1,131 @@
+    {
+        public TentacleConfigurationsAttribute(
+            bool testCommonVersions = false,
+            bool testCapabilitiesServiceInterestingVersions = false,
+            bool testNoCapabilitiesServiceVersions = false,
+            bool testScriptIsolationLevelVersions = false,
+            params object[] additionalParameterTypes)
+            : base(
+                typeof(TentacleConfigurationTestCases),
+                nameof(TentacleConfigurationTestCases.GetEnumerator),
+                new object[] {testCommonVersions, testCapabilitiesServiceInterestingVersions, testNoCapabilitiesServiceVersions, testScriptIsolationLevelVersions, additionalParameterTypes})
+        {
+        }
+    }
+
+    static class TentacleConfigurationTestCases
+    {
+        public static IEnumerator GetEnumerator(
+            bool testCommonVersions,
+            bool testCapabilitiesInterestingVersions,
+            bool testNoCapabilitiesServiceVersions,
+            bool testScriptIsolationLevel,
+            object[] additionalParameterTypes)
+        {
+            var tentacleTypes = new[] {TentacleType.Listening, TentacleType.Polling};
+            var halibutTypes = new[] {SyncOrAsyncHalibut.Sync, SyncOrAsyncHalibut.Async};
+            List<Version?> versions = new();
+
+            if (testCommonVersions)
+            {
+                versions.AddRange(new[]
+                {
+                    TentacleVersions.Current,
+                    TentacleVersions.v5_0_15_LastOfVersion5,
+                    TentacleVersions.v6_3_417_LastWithScriptServiceV1Only,
+                    TentacleVersions.v7_0_1_ScriptServiceV2Added
+                });
+            }
+
+            if (testCapabilitiesInterestingVersions)
+            {
+                versions.AddRange(new[]
+                {
+                    TentacleVersions.Current,
+                    TentacleVersions.v5_0_4_FirstLinuxRelease,
+                    TentacleVersions.v5_0_12_AutofacServiceFactoryIsInShared,
+                    TentacleVersions.v6_3_417_LastWithScriptServiceV1Only, // the autofac service is in tentacle, but tentacle does not have the capabilities service.
+                    TentacleVersions.v7_0_1_ScriptServiceV2Added
+                });
+            }
+
+            if (testScriptIsolationLevel)
+            {
+                versions.AddRange(new[]
+                {
+                    TentacleVersions.Current,
+                    TentacleVersions.v6_3_417_LastWithScriptServiceV1Only,
+                    TentacleVersions.v7_0_1_ScriptServiceV2Added // Testing against v1 and v2 script services
+                });
+            }
+
+            if (testNoCapabilitiesServiceVersions)
+            {
+                versions.AddRange(new []
+                {
+                    TentacleVersions.v6_3_451_NoCapabilitiesService
+                });
+            }
+
+            var testCases =
```

Comment:
If we don't have a version, will this result in no test cases?

I saw in `ClientFileTransferRetriesTimeout` that we do not choose a version. And I guess we shouldn't as it does not want to test versions.

But since all our version booleans are false by default, it means that `versions` will be empty in that case.

I did a little (and very ugly) test in LinqPad to see what would happen if we did a join with an empty list. The following results in 0 results:

```
var x = new[]{"One", "Two", "Three"};
var y = new[] { 1, 2, 3 };
var z = new List<string?>();


var r = 
from a in x
from b in y
from c in z
select new {
	a,b,c
};

r.ToList().Dump();
```
When we add something to `z`, it lights up like a Christmas tree and does what we expect.

I could be missing something, but if feels like maybe we should add `TentacleVersions.Current` if nothing else sets the version?

Reply:
If we haven't already, maybe it'll be worth getting a test output from TeamCity from master, and compare it to a TeamCity build for this PR, and ensure the tests line up?